src/pages/Pool/RemoveLiquidity.tsx

```diff
@@ -389,68 +400,109 @@ export default function RemoveLiquidity({ token0, token1 }: { token0: string; to
       message
     })
 
-    library.send('eth_signTypedData_v4', [account, data]).then(_signature => {
-      const signature = splitSignature(_signature)
-      setSigInputs([signature.v, signature.r, signature.s])
-      setSigned(true)
-    })
+    library
+      .send('eth_signTypedData_v4', [account, data])
+      .then(splitSignature)
+      .then(signature => {
+        setSignatureData({
+          v: signature.v,
+          r: signature.r,
+          s: signature.s,
+          deadline: deadlineForSignature
+        })
+      })
+      .catch(error => {
+        // for all errors other than 4001 (EIP-1193 user rejected request), fall back to manual approve
+        if (error?.code !== 4001) {
+          approveCallback()
+        }
+      })
   }
 
   function resetModalState() {
-    setSigned(false)
-    setSigInputs(null)
+    setSignatureData(null)
     setAttemptedRemoval(false)
     setPendingConfirmation(true)
   }
 
   async function onRemove() {
     setAttemptedRemoval(true)
+
     const router = getRouterContract(chainId, library, account)
-    let method, args, estimate
-
-    // removal with ETH
-    if (tokens[Field.TOKEN0].equals(WETH[chainId]) || tokens[Field.TOKEN1].equals(WETH[chainId])) {
-      method = router.removeLiquidityETHWithPermit
-      estimate = router.estimateGas.removeLiquidityETHWithPermit
-
-      const token0IsETH = tokens[Field.TOKEN0].equals(WETH[chainId])
-
-      args = [
-        tokens[token0IsETH ? Field.TOKEN1 : Field.TOKEN0].address,
-        parsedAmounts[Field.LIQUIDITY].raw.toString(),
-        calculateSlippageAmount(
-          parsedAmounts[token0IsETH ? Field.TOKEN1 : Field.TOKEN0],
-          ALLOWED_SLIPPAGE
-        )[0].toString(),
-        calculateSlippageAmount(
-          parsedAmounts[token0IsETH ? Field.TOKEN0 : Field.TOKEN1],
-          ALLOWED_SLIPPAGE
-        )[0].toString(),
-        account,
-        deadline,
-        false,
-        sigInputs[0],
-        sigInputs[1],
-        sigInputs[2]
-      ]
+
+    const token0IsETH = tokens[Field.TOKEN0].equals(WETH[chainId])
+    const oneTokenIsETH = token0IsETH || tokens[Field.TOKEN1].equals(WETH[chainId])
+
+    let estimate, method, args
+    // we have approval, use normal remove liquidity
+    if (mustApprove === false) {
```

Comment:
false for undefined, but seems intentional

Reply:
yeah this is a bit of a weird one, our disabled states should prevent `mustApprove` from ever being `undefined` at this line